### PR TITLE
Updates to labels along lines

### PR DESCRIPTION
**Pull request by nickidlugash**
- Removes anchors that are too close to another anchor with the same text
- Takes into account icon size for calculating label length
- Recalculates spacing for long labels
- Adjusts offsets to first anchors if line is continued from outside the tile boundary

### CODE REVIEW

File: debug/site.js [hunk 1]
```diff
@@ -5,11 +5,13 @@ var map = new mapboxgl.Map({
     container: 'map',
     zoom: 12.5,
     center: [38.888, -77.01866],
-    style: 'https://www.mapbox.com/mapbox-gl-styles/styles/bright-v7.json',
```
Comment (ansis) [inline debug/site.js]: can you undo the changes to this file?

**Comment by mourner** [comment]
Awesome! Should be squashed into one commit and rebased on master.

**Comment by ansis** [comment]
:tada:

**Comment by jfirebaugh** [comment]
Nice work @nickidlugash!
